hutch_python/tests/conftest.py

```diff
@@ -33,29 +35,55 @@ def restore_logging():
     logging.root.handlers = prev_handlers
 
 
-Experiment = namedtuple('Experiment', ('run', 'proposal'))
+Experiment = namedtuple('Experiment', ('run', 'proposal',
+                                       'user', 'pw', 'kerberos'))
 
 
 class QSBackend:
-    def __init__(self, run, proposal):
+    def __init__(self, run, proposal, use_kerberos=True, user=None, pw=None):
         self.run = run
         self.proposal = proposal
+        self.user = user
+        self.pw = pw
+        self.kerberos = use_kerberos
 
     def find(self, multiples=False, **kwargs):
         devices = [{
             '_id': 'TST:USR:MMN:01',
             'beamline': 'TST',
             'device_class': 'hutch_python.tests.conftest.Experiment',
             'location': 'Hutch-main experimental',
-            'args': ['{{run}}', '{{proposal}}'],
+            'args': ['{{run}}', '{{proposal}}',
+                     '{{user}}', '{{pw}}', '{{kerberos}}'],
             'kwargs': {},
             'name': 'inj_x',
             'prefix': 'TST:USR:MMN:01',
             'purpose': 'Injector X',
             'type': 'Device',
             'run': self.run,
+            'user': self.user,
+            'pw': self.pw,
```

Comment:
It's moderately alarming if these are stored somewhere in the device info.

Reply:
This is all for testing **only**. The issue is that the `QSBackend` is a local variable in `Plugin.get_objects()`. It is therefore hard to introspect and see that it was setup properly. To get around this I made a fake object that stores the password it was instantiated that way I can just check the output of `get_objects` for the state of the client. In practice, this information is dropped after `get_objects` as they are only local variables.

Reply:
Sounds good, just looked suspect when I was reading down the line.